misc/i18n/generateTranslations.ts

```diff
@@ -1,28 +1,26 @@
+//@ts-ignore
 import { writeFile } from "fs"
+//@ts-ignore
 import { readFile } from "fs/promises"
 import yaml from "js-yaml"
 import { TsUtils } from '../../src/lib/TsUtils'
 import { betaInformation } from "./betaInformation"
 import pathConfig from './AvatarBaseType.json'
 import AvatarConfig from './AvatarConfig.json'
-import rankConfig from './AvatarRankConfig.json'
-import skillConfig from './AvatarSkillConfig.json'
 import damageConfig from './DamageType.json'
 import lightconeConfig from './EquipmentConfig.json'
-import lightconeRankConfig from './EquipmentSkillConfig.json'
 import relicSetConfig from './RelicSetConfig.json'
 import relicEffectConfig from './RelicSetSkillConfig.json'
 
 const precisionRound = TsUtils.precisionRound
 
-const Locales = ['zh', 'de', 'en', 'es', 'fr', 'id', 'ja', 'ko', 'pt', 'ru', 'th', 'vi']
+const inputLocales = ['zh', 'de', 'en', 'es', 'fr', 'id', 'ja', 'ko', 'pt', 'ru', 'th', 'vi'] as const
 
-const TrailblazerPaths = ['Warrior', 'Knight', 'Shaman']
+const outputLocales = [...inputLocales, 'it'] as const
 
-const multiPathIds = [1001, 1224]
-
-const outputLocalesMapping = {
-  zh: ['zh'],
+// keys must correspond to an available textmap, values are the output locales for a given textmap
+// e.g. the english textmap is used for both the english and italian gameData files
+const outputLocalesMapping: {[key in typeof inputLocales[number]]: typeof outputLocales[number][]} = {
```

Comment:
After learning about Record<> type syntax I've been preferring that over key maps for readability:

```
const outputLocalesMapping: Record<typeof inputLocales[number], typeof outputLocales[number][]> = {
...
```

Even better, since `[key in typeof inputLocales[number]` is used multiple times we can simplify as this and reuse it in TB_NAMES / Overrides

```

type InputLocale = typeof inputLocales[number]
type OutputLocale = typeof outputLocales[number]

// keys must correspond to an available textmap, values are the output locales for a given textmap
// e.g. the english textmap is used for both the english and italian gameData files
const outputLocalesMapping: Record<InputLocale, OutputLocale[]> = {
```